### PR TITLE
Add datasource hosting_privatedatabase

### PR DESCRIPTION
**Pull request by wiseflat**
Add support for hosting private database.

### CODE REVIEW

File: ovh/data_hosting_privatedatabase_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+					resource.TestCheckResourceAttr(
+						"data.ovh_hosting_privatedatabase.database",
+						"cpu",
+						"1",
+					),
+					resource.TestCheckResourceAttr(
+						"data.ovh_hosting_privatedatabase.database",
+						"datacenter",
+						"gra3",
+					),
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase_test.go:28]: these values are to specific to your own database. you should maybe only test that the attribute is set, and not check its value

File: ovh/data_hosting_privatedatabase_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+					resource.TestCheckResourceAttr(
+						"data.ovh_hosting_privatedatabase.database",
+						"offer",
+						"public",
+					),
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase_test.go:33]: idem

File: ovh/data_hosting_privatedatabase_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+					resource.TestCheckResourceAttr(
+						"data.ovh_hosting_privatedatabase.database",
+						"type",
+						"postgresql",
+					),
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase_test.go:38]: idem

File: ovh/data_hosting_privatedatabase_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+					resource.TestCheckResourceAttr(
+						"data.ovh_hosting_privatedatabase.database",
+						"version_number",
+						"12",
+					),
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase_test.go:43]: idem

File: ovh/data_hosting_privatedatabase.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	d.Set("version", ds.Version)
+	d.Set("version_label", ds.VersionLabel)
+	d.Set("version_number", ds.VersionNumber)
+
+	if err != nil {
+		return fmt.Errorf(
+			"Error reading Hosting PrivateDatabase for %s: %q",
+			serviceName,
+			err,
+		)
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase.go:161]: to be removed, error is already checked earlier

File: ovh/data_hosting_privatedatabase.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	err := config.OVHClient.Get(
+		fmt.Sprintf(
+			"/hosting/privateDatabase/%s",
+			url.PathEscape(serviceName),
+		),
+		&ds,
+	)
+
+	if err != nil {
+		return fmt.Errorf(
```
Comment (yanndegat) [inline ovh/data_hosting_privatedatabase.go:128]: you should check 404 specifically

File: ovh/data_hosting_privatedatabase_test.go (hunk 1 above)
Comment (wiseflat) [reply ovh/data_hosting_privatedatabase_test.go:26]: fixed

File: ovh/data_hosting_privatedatabase_test.go (hunk 2 above)
Comment (wiseflat) [reply ovh/data_hosting_privatedatabase_test.go:30]: fixed

File: ovh/data_hosting_privatedatabase_test.go (hunk 3 above)
Comment (wiseflat) [reply ovh/data_hosting_privatedatabase_test.go:34]: fixed

File: ovh/data_hosting_privatedatabase_test.go (hunk 4 above)
Comment (wiseflat) [reply ovh/data_hosting_privatedatabase_test.go:38]: fixed

File: ovh/data_hosting_privatedatabase.go (hunk 5 above)
Comment (wiseflat) [reply ovh/data_hosting_privatedatabase.go]: fixed

**Comment by wiseflat** [comment]
I close this PR and I will reopen it with all resources/datasources of the private database features.
Thanks

**Pull request closed without merging**